### PR TITLE
Fixes #2179 - Port capnp serialization to AnomalyRegion class

### LINKED ISSUE #2179: Port capnp serialization to AnomalyRegion class.
See parent issue: https://github.com/numenta/nupic/issues/2158

### PR DESCRIPTION
**Pull request by david-ragazzi**
Fixes https://github.com/numenta/nupic/issues/2179

### CODE REVIEW

File: nupic/regions/AnomalyRegion.capnp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@0xc065162b7aa5ede9;
+
+# Next ID: ???
+struct AnomalyRegionProto {
+  version @0 :UInt16;
```
Comment (scottpurdy) [inline nupic/regions/AnomalyRegion.capnp]: remove this

File: nupic/regions/AnomalyRegion.py [hunk 2]
```diff
@@ -78,6 +78,17 @@ def __init__(self, *args, **kwargs):
     self.prevPredictedColumns = numpy.zeros([], dtype="float32")
 
 
+  @classmethod
+  def read(cls, proto):
+    anomRegion = object.__new__(cls)
```
Comment (scottpurdy) [inline nupic/regions/AnomalyRegion.py]: anomalyRegion

Comment (david-ragazzi) [inline nupic/regions/AnomalyRegion.py]: > anomalyRegion

Done.

File: nupic/regions/AnomalyRegion.capnp (hunk 1 above)
Comment (david-ragazzi) [inline nupic/regions/AnomalyRegion.capnp]: > remove this

Done.

**Comment by david-ragazzi** [comment]
@scottpurdy All changes done. This PR is ready but depends of the global assert function (https://github.com/numenta/nupic/pull/2197).

File: nupic/regions/AnomalyRegion.py [hunk 3]
```diff
@@ -78,6 +78,17 @@ def __init__(self, *args, **kwargs):
...
 
+  @classmethod
+  def read(cls, proto):
+    anomalyRegion = object.__new__(cls)
+    anomalyRegion.prevPredictedColumns = numpy.array(proto.prevPredictedColumns)
+    return anomalyRegion
+
+
+  def write(self, proto):
+    proto.prevPredictedColumns = self.prevPredictedColumns.tolist()
```
Comment (scottpurdy) [inline nupic/regions/AnomalyRegion.py]: Does this work? Don't you need to first init the proto list and then set it?

http://jparyani.github.io/pycapnp/quickstart.html#list

File: tests/unit/nupic/regions/anomaly_region_test.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2014, Numenta, Inc.  Unless you have purchased from
```
Comment (scottpurdy) [inline tests/unit/nupic/regions/anomaly_region_test.py]: year

File: tests/unit/nupic/regions/anomaly_region_test.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import tempfile
+import unittest2 as unittest
```
Comment (scottpurdy) [inline tests/unit/nupic/regions/anomaly_region_test.py]: just `import unittest`

**Comment by scottpurdy** [comment]
:+1:

File: nupic/regions/AnomalyRegion.py (hunk 3 above)
Comment (david-ragazzi) [inline nupic/regions/AnomalyRegion.py]: > Don't you need to first init the proto list and then set it?

Not, not longer. 

https://groups.google.com/forum/#!topic/capnproto/fFxJp7D-99I

**Comment by david-ragazzi** [comment]
@scottpurdy Changes done. But I can't merge while you don't answer my question in https://github.com/numenta/nupic/pull/2197

**Comment by david-ragazzi** [comment]
@scottpurdy Now I'm using the defaul assert function for this PR. The test has to temporaly convert numpy arrays to list because default assert function doesn't work (well) with numpy.

**Pull request merged**